envoy/ssl/handshaker.h

```diff
@@ -231,7 +232,7 @@ class TlsCertificateSelector {
    * @return context will have the same lifetime as ``ServerContextImpl``.
    */
   virtual std::pair<const Ssl::TlsContext&, OcspStapleAction>
-  findTlsContext(absl::string_view sni, bool client_ecdsa_capable, bool client_ocsp_capable,
+  findTlsContext(absl::string_view sni, absl::optional<std::vector<int>> client_ecdsa_capabilies, bool client_ocsp_capable,
```

Comment:
I think this type can be (and everywhere else) just `std::vector<int>`. An empty vector is good enough for no ecdsa support.